src/UI/class-admin-columns-analytics.php

```diff
@@ -0,0 +1,322 @@
+				<?php echo esc_html( get_formatted_number( $visitors ) . ' ' . _n( 'visitor', 'visitors', $visitors, 'wp-parsely' ) ); ?>
+			</span> <br/>
+
+			<?php
+		}
+
+		if ( isset( $metrics['avg_engaged'] ) ) {
+			$engaged_seconds = $metrics['avg_engaged'] * 60;
+			?>
+
+			<span class='parsely-post-avg_engaged'>
+				<?php echo esc_html( get_formatted_time( $engaged_seconds ) . ' ' . __( 'avg time', 'wp-parsely' ) ); ?>
+			</span> <br/>
+
+			<?php
+		}
+	}
+
+	/**
+	 * Get publish date params for analytics API.
+	 *
+	 * For getting date params we loop through all the posts and calculate
+	 * the min and max publish dates based on which we will limit the analytics API.
+	 *
+	 * @param WP_Post[] $posts Array of post objects.
+	 *
+	 * @return Analytics_Post_API_Params|null
+	 */
+	private function get_publish_date_params_for_analytics_api( array $posts ) {
+		if ( count( $posts ) === 0 ) {
+			return null;
+		}
+
+		$max_date_time = '';
+		$min_date_time = '';
+
+		foreach ( $posts as $post ) {
+			$published_date_time = $post->post_date_gmt;
+
+			if ( '' === $min_date_time || $published_date_time < $min_date_time ) {
+				$min_date_time = $published_date_time;
+			}
+
+			if ( $published_date_time > $max_date_time ) {
+				$max_date_time = $published_date_time;
+			}
+		}
+
+		return array(
+			'pub_date_start' => ( new DateTime( $min_date_time ) )->format( DATE_UTC_FORMAT ),
+			'pub_date_end'   => ( new DateTime( $max_date_time ) )->format( DATE_UTC_FORMAT ),
+		);
+	}
+
+	/**
+	 * Get unique key which we can use for Parse.ly stats map.
+	 * Mainly we need this because Parse.ly doesn't have anything unique in its analytics API.
+	 *
+	 * @param Analytics_Post $analytics_post Post analytics obj returned from Parse.ly API.
+	 *
+	 * @return string
+	 */
+	private function get_unique_stats_key_from_analytics( $analytics_post ): string {
+		if ( ! isset( $analytics_post['title'] ) || ! isset( $analytics_post['pub_date'] ) ) {
+			return '';
+		}
+
+		return $analytics_post['title'] . '-' . $analytics_post['pub_date'];
+	}
+
+	/**
+	 * Get unique key from currently set post which we can use to get data from Parse.ly stats map.
+	 *
+	 * @return string
+	 */
+	private function get_unique_stats_key_of_current_post(): string {
+		global $post;
+
+		$published_date     = $post->post_date_gmt;
+		$published_utc_date = ( new DateTime( $published_date ) )->format( DATE_TIME_UTC_FORMAT );
+
+		return get_the_title() . '-' . $published_utc_date;
+	}
+
+	/**
+	 * Return TRUE if the current screen is post list screen.
+	 *
+	 * @since 3.7.0
+	 * @todo Add support for pages and custom post types (currently pending due to API limitation).
+	 *
+	 * @return bool
+	 */
+	private function is_post_list_screen(): bool {
+		if ( ! isset( $_SERVER['REQUEST_URI'] ) ) {
+			return false;
+		}
+
+		$page_url = esc_url_raw( $_SERVER['REQUEST_URI'] );
+
+		return '/wp-admin/edit.php' === $page_url || strpos( $page_url, 'post_type=post' ) !== false;
```

Comment:
You can't depend on this hardcoded URL check. WordPress can be installed to any path. Use `get_current_screen`

Reply:
I tried using `get_current_screen` but it returns `null` ..... still experimenting with this.